### PR TITLE
[2.13.x] DDF-3941 Fix Metacard ID not being set in CatalogApplication

### PR DESCRIPTION
**Pull request by blen-desta**
#### What does this PR do?
Fixes Metacard ID not being set as a header in the CatalogApplication.

#### Who is reviewing it? 
@garrettfreibott  @brjeter 

#### Ask 2 committers to review/merge the PR and tag them here.
@andrewkfiedler
@rzwiefel

#### How should this be tested?
Ingest data from Intrigue with the network tab open and verify that a location header is returned in the response.

#### What are the relevant tickets?
[DDF-3941](https://codice.atlassian.net/browse/DDF-3941)

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/catalog/CatalogApplication.java [hunk 1]
```diff
@@ -372,7 +372,8 @@ private Response addDocument(
       URI uri = new URI(requestUrl.toString());
       UriBuilder uriBuilder = new UriBuilderImpl(uri).path("/" + id);
 
-      res.redirect(uriBuilder.build().toString(), HttpStatus.SC_CREATED);
+      res.status(HttpStatus.SC_CREATED);
+      res.header("Location", uriBuilder.build().toString());
       res.header(Metacard.ID, id);
```
Comment (brjeter) [inline catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/catalog/CatalogApplication.java]: ❓ Could you give a bit of background on the reason for this Location header being added? This was not added in the old RestEndpoint and isn't in the new one either, so why is this necessary now?

Comment (blen-desta) [reply catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/catalog/CatalogApplication.java]: It is added in the RESTEndpoint. https://github.com/codice/ddf/blob/3cc3d8c027cdde26936765490fb58cf969ed672f/catalog/rest/catalog-rest-endpoint/src/main/java/org/codice/ddf/endpoints/rest/RESTEndpoint.java#L353

`res.redirect` added the location but for some reason the `res.header(Metacard.ID, id);` line under it wasn't getting called.

Comment (brjeter) [reply catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/catalog/CatalogApplication.java]: Oops I missed that in the `RestEndpoint`. It could be that the `redirect` method call is meant to be a terminating call. Not sure.

**Comment by blen-desta** [comment]
build now
